site/en/docs/web-platform/navigation-api/index.md

```diff
@@ -92,90 +108,253 @@ Links might come and go on your page, and they're not the only way users can nav
+
+A way to delay the URL change is being [discussed on GitHub](https://github.com/WICG/navigation-api/issues/66), but it's generally recommended to immediately update the page with some sort of placeholder for the incoming content:
 
 ```js
-navigation.addEventListener('navigateerror', event => {
-  loadingIndicator.hidden = true; // also hide indicator
-  showMessage(`Failed to load page: ${event.message}`);
+navigation.addEventListener('navigate', navigateEvent => {
+  if (shouldNotIntercept(navigateEvent)) return;
+  const url = new URL(navigateEvent.destination.url);
+
+  if (url.pathname.startsWith('/articles/')) {
+    navigateEvent.intercept({
+      async handler() {
+        // The URL has already changed, so quickly show a placeholder.
+        renderArticlePagePlaceholder();
+        // Then fetch the real data.
+        const articleContent = await getArticleContent(url.pathname);
+        renderArticlePage(articleContent);
+      },
+    });
+  }
 });
 ```
 
-The "navigateerror" event listener, which receives an `ErrorEvent`, is particularly handy as it's guaranteed to receive any errors from your code that's setting up a new page.
-You can simply `await fetch()` knowing that if the network is unavailable, the error will eventually be routed to `"navigateerror"`.
+This not only avoids URL resolution issues, it also feels fast because you're instantly responding to the user.
+
+### Abort signals
 
-### Abort Signals
+Since you're able to do asynchronous work in an `intercept()` handler, it's possible for the navigation to become redundant.
+This happens when:
 
-Since you're able to do asynchronous work while preparing a new page, it's possible that the transition your code is handling (to load a specific URL or state) might get preempted, or considered out-of-date.
-This might happen because the user just clicked on another link, or your code performs another navigation.
+- The user clicks another link, or some code performs another navigation.
+  In this case the old navigation is abandoned in favour of the new navigation.
+- The user clicks the 'stop' button in the browser.
 
-To deal with any of these possibilities, the event passed to the "navigate" listener contains a `signal` property, which is an `AbortSignal`.
+To deal with any of these possibilities, the event passed to the `"navigate"` listener contains a `signal` property, which is an `AbortSignal`.
 For more information see [Abortable fetch][abortable-fetch].
+
 The short version is it basically provides an object that fires an event when you should stop your work.
 Notably, you can pass an `AbortSignal` to any calls you make to `fetch()`, which will cancel in-flight network requests if the navigation is preempted.
 This will both save the user's bandwidth, and reject the `Promise` returned by `fetch()`, preventing any following code from e.g., updating the DOM to show a now invalid page navigation.
```

Comment:
```suggestion
This will both save the user's bandwidth, and reject the `Promise` returned by `fetch()`, preventing any following code from actions such as updating the DOM to show a now invalid page navigation.
```